src/test/resources/overbroadConstraintsTest/suggests_star_in_complex_situations

```diff
@@ -0,0 +1,17 @@
+versions.props:
+com.example.*:*= 1.0.0
+
+versions.lock:
+com.example.core:artifact-random:1.0.0 (2 constraints: abcdef1)
+com.example.module:artifact-platform-commons:2.0.0 (2 constraints: abcdef1)
+com.example.module:artifact-platform-new:2.0.0 (2 constraints: abcdef1)
+com.example.different:so-very-different:2.0.0 (2 constraints: abcdef1)
+com.example.module:artifact-different:3.0.0 (2 constraints: abcdef1)
+com.example.module:artifact-different-again:4.0.0 (2 constraints: abcdef1)
+
+new pins (blank when no pins needed):
```

Comment:
The new code is better in it's new pin suggestions - but it's deleting half the props in the versions.props when I run the fixer on some internal projects. I think all the ones where the version.lock items have all the same versions.

I think we should perhaps test a little higher level - have the output here be a diff between the original verisons.props and the new versions.props (I think if you name the file `.diff` github will even syntax highlight it). This way we can test the deleting/inserting code too.

Reply:
which internal projects are causing issues so I can do some quick test

Reply:
Cool - I should have fixed this bug and have ensure that the tests cover it now - I have also switch to the diff method for the tests